src/crud_internal.c

```diff
@@ -119,188 +124,174 @@ int createObject( wrp_msg_t *reqMsg , wrp_msg_t **response)
 		}
 	}
 
-	if(reqMsg->u.crud.dest !=NULL)
+	if(jsonData !=NULL)
 	{
-		destVal = strdup(reqMsg->u.crud.dest);
-		ParodusInfo("destVal is %s\n", destVal);
-
-		if( (destVal != NULL))
-		{
-			child_ptr = strtok(destVal , "/");
+		free( jsonData );
+		jsonData = NULL;
+	}
 
-			/* Get the 1st object */
-			obj[0] = strdup( child_ptr );
-			ParodusPrint( "parent is %s\n", obj[0] );
+	if(reqMsg->u.crud.dest !=NULL)
+	{
+		ParodusInfo("reqMsg->u.crud.dest is %s\n", reqMsg->u.crud.dest);
+		child_ptr = strtok(reqMsg->u.crud.dest , "/");
 
-			free(destVal);
+		/* Get the 1st object */
+		obj[0] = strdup( child_ptr );
+		ParodusPrint( "parent is %s\n", obj[0] );
 
-			while( child_ptr != NULL)
+		while( child_ptr != NULL)
+		{
+			child_ptr = strtok( NULL, "/" );
+			if( child_ptr != NULL)
 			{
-				child_ptr = strtok( NULL, "/" );
-				if( child_ptr != NULL)
-				{
-					obj[i] = strdup( child_ptr );
-					ParodusPrint( "child obj[%d]:%s\n", i, obj[i] );
-					i++;
-				}
+				obj[i] = strdup( child_ptr );
+				ParodusPrint( "child obj[%d]:%s\n", i, obj[i] );
+				i++;
 			}
+		}
 
-			objlevel = i;
-			ParodusInfo( "Number of object level %d\n", objlevel );
+		objlevel = i;
+		ParodusInfo( "Number of object level %d\n", objlevel );
 
-			/* Valid request will be mac:14cfexxxx/parodus/tags/${name} which is objlevel 4 */
-			if(objlevel == 4)
+		/* Valid request will be mac:14cfexxxx/parodus/tags/${name} which is objlevel 4 */
+		if(objlevel == 4)
```

Comment:
Should add verification check for `0 == strcmp(obj[1], "mac:112233445566")`, `0 == strcmp(obj[2], "parodus")`, `0 == strcmp(obj[3], "tags")` unless the check is done elsewhere.

Reply:
Currently this check is done internally once objlevel is valid.